packages/flutter_tools/lib/src/commands/upgrade.dart

```diff
@@ -222,10 +228,81 @@ class UpgradeCommandRunner {
     }
   }
 
+  /// Checks if the Flutter git repository is tracking a "standard remote".
+  ///
+  /// Using `flutter upgrade` is not supported from a non-standard remote. A git
+  /// remote should have the same url as [_flutterGit] to be considered as a
+  /// "standard" remote.
+  ///
+  /// Exits tool if the tracking remote is not standard.
+  void verifyStandardRemote(FlutterVersion localVersion) {
+    // If localVersion.repositoryUrl is null, exit
+    if (localVersion.repositoryUrl == null) {
+      throwToolExit(
+        'Unable to upgrade Flutter: The Flutter SDK is tracking an "unknown" '
+        'remote.\n'
+        'Re-install Flutter by going to $_flutterInstallDocs. Alternatively, use '
```

Comment:
Remove "Alternatively, use '
        '"git" directly to configure a valid upstream for the branch and retry"

Reply:
Done.